nucliadb/nucliadb/ingest/orm/shard.py

```diff
@@ -86,13 +98,32 @@ async def add_resource(
             await indexing.index(indexpb, shardreplica.node)
 
             try:
-                res: Counter = await NODES[shardreplica.node].sidecar.GetCount(shardreplica.shard)  # type: ignore
-                if count < res.resources:
-                    count = res.resources
-            except Exception:
-                pass
-
-        return count
+                counter: Counter = await NODES[shardreplica.node].sidecar.GetCount(shardreplica.shard)  # type: ignore
+                shard_counter = ShardCounter(
+                    shard=self.sharduuid,
+                    fields=counter.resources,
+                    paragraphs=counter.paragraphs,
+                )
+                if last_counter is not None and self.counters_are_different(
+                    last_counter, shard_counter
+                ):
+                    with sentry_sdk.push_scope() as scope:
+                        scope.set_extra("shard", [self.sharduuid])
+                        scope.set_extra(
+                            "paragraphs",
+                            [shard_counter.paragraphs, last_counter.paragraphs],
+                        )
+                        scope.set_extra(
+                            "fields", [shard_counter.fields, last_counter.fields]
+                        )
+                        sentry_sdk.capture_message(
+                            f"Detected shard replicas out of sync"
+                        )
+                last_counter = shard_counter
+            except Exception as exc:
+                sentry_sdk.capture_exception(exc)
```

Comment:
Can you put this logic inside the `if SENTRY:` condition?

Reply:
Yep